test/client_test.dart

```diff
@@ -0,0 +1,410 @@
+          'X-Other-Header': 'Other Value',
+          'User-Agent': userAgent()
+        },
+      );
+      var body = await response.readAsString();
+
+      expect(response.statusCode, equals(200));
+      expect(
+          body,
+          parse(equals({
+            'method': 'PATCH',
+            'path': '/',
+            'headers': {
+              'content-type': [
+                'application/x-www-form-urlencoded; charset=utf-8'
+              ],
+              'content-length': ['40'],
+              'user-agent': [userAgent()],
+              'x-random-header': ['Value'],
+              'x-other-header': ['Other Value']
+            },
+            'body': 'some-field=value&other-field=other+value'
+          })));
+    });
+
+    // DELETE
+
+    test('delete', () async {
+      var response = await platformClient().delete(serverUrl, headers: {
+        'X-Random-Header': 'Value',
+        'X-Other-Header': 'Other Value',
+        'User-Agent': userAgent()
+      });
+      var body = await response.readAsString();
+
+      expect(response.statusCode, equals(200));
+      expect(
+          body,
+          parse(equals({
+            'method': 'DELETE',
+            'path': '/',
+            'headers': {
+              'user-agent': [userAgent()],
+              'x-random-header': ['Value'],
+              'x-other-header': ['Other Value']
+            }
+          })));
+    });
+
+    // READ
+
+    test('read', () async {
+      var body = await platformClient().read(serverUrl, headers: {
+        'X-Random-Header': 'Value',
+        'X-Other-Header': 'Other Value',
+        'User-Agent': userAgent()
+      });
+
+      expect(
+          body,
+          parse(equals({
+            'method': 'GET',
+            'path': '/',
+            'headers': {
+              'user-agent': [userAgent()],
+              'x-random-header': ['Value'],
+              'x-other-header': ['Other Value']
+            }
+          })));
+    });
+
+    test('read throws an error for a 4** status code', () async {
+      platformClient().read(serverUrl.resolve('/error')).then((_) {
+        expect(true, isFalse);
+      }).catchError((err) {
+        expect(err, isClientException);
+      });
+    });
+
+    test('readBytes', () async {
+      var body = await platformClient().readBytes(serverUrl, headers: {
+        'X-Random-Header': 'Value',
+        'X-Other-Header': 'Other Value',
+        'User-Agent': userAgent()
+      });
+
+      expect(
+          new String.fromCharCodes(body),
+          parse(equals({
+            'method': 'GET',
+            'path': '/',
+            'headers': {
+              'user-agent': [userAgent()],
+              'x-random-header': ['Value'],
+              'x-other-header': ['Other Value']
+            }
+          })));
+    });
+
+    test('readBytes throws an error for a 4** status code', () {
+      platformClient().readBytes(serverUrl.resolve('/error')).then((_) {
```

Comment:
Is there a specific way I should be using one of the `expect` functions? I wasn't able to get it to work otherwise.